### PR TITLE
Add examples/Makefile

### PR DESCRIPTION
**Pull request by masutaka**
terraform-provider-healthchecksio is not a official provider of terraform. So I believe this Makefile will help this provider users :smile:

:warning: Currently it's only support macOS.

### CODE REVIEW

File: examples/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# WIP: Currently this is only support macOS.
+
+OS_NAME := $(shell uname -s)
+ARCH := $(shell uname -m)
+DOWNLOAD_DIR := $(shell mktemp -d)
+PLUGIN_DIR := .terraform/plugins/$(shell echo $(OS_NAME) | tr A-Z a-z)_amd64 # hard coding
```
Comment (masutaka) [inline examples/Makefile:6]: I could not know how to get `amd64` on macOS.

I wanted to get the value of `DefaultPluginVendorDir`. But It seems the API doesn't exist. Only Terraform knows this directory.

* https://github.com/hashicorp/terraform/blob/v0.11.8/command/command.go#L22-L28
* https://www.terraform.io/docs/configuration/providers.html#third-party-plugins

**Comment by masutaka** [comment]
FYI: Just added to https://healthchecks.io/docs/resources/ :tada:

See also the thread https://twitter.com/masutaka/status/1048942095765929986

**Comment by kristofferahl** [comment]
Looks good. If it's for building binaries, have you tried using goreleaser?
That is what we use for creating github releases but it can also be used for only creating the binaries I believe. Have a look at .goreleaser.yml in the root directory of this repository.

https://goreleaser.com/

I will approve this PR and you can decide if it should still be merged. 😃

**Review by kristofferahl — APPROVED** [review]

**Comment by kristofferahl** [comment]
@masutaka After installing goreleaser, you can build binaries by running the following command.

```bash
goreleaser --rm-dist --skip-publish
```

The binaries will be created in the `dist` directory.

**Comment by masutaka** [comment]
My description might not have been enough.

Use `$ make init` instead of `$ terraform init`.

```
$ make init
curl -sL https://github.com/kristofferahl/terraform-provider-healthchecksio/releases/download/v1.1.0/terraform-provider-healthchecksio_1.1.0_Darwin_x86_64.tar.gz | tar xz -C /var/folders/35/z_r44myd0td8p40zr3t3wzph0000gn/T/tmp.E7TOQjDU
mkdir -p .terraform/plugins/darwin_amd64
cp /var/folders/35/z_r44myd0td8p40zr3t3wzph0000gn/T/tmp.E7TOQjDU/terraform-provider-healthchecksio_v1.1.0 .terraform/plugins/darwin_amd64/terraform-provider-healthchecksio_v1.1.0
terraform init

Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
rm -rf /var/folders/35/z_r44myd0td8p40zr3t3wzph0000gn/T/tmp.E7TOQjDU
```

You can download the binary with the following directory structure;

```
$ tree .terraform
.terraform
├── plugins
│   └── darwin_amd64
│       ├── lock.json
│       └── terraform-provider-healthchecksio_v1.1.0
└── terraform.tfstate

2 directories, 3 files
```

`terraform-provider-healthchecksio` is not a official provider of terraform. So this Makefile may be helpful to users.

----

Also I tried goreleaser. I needed `--skip-validate` option.

```
$ goreleaser --rm-dist --skip-publish
(snip)
   ⨯ release failed after 0.59s error=git tag v1.1.0 was not made against commit 0f59a82a82c479aaf2750f467afa74554916599d


$ goreleaser --rm-dist --skip-publish --skip-validate
(snip)
   • release succeeded after 26.10s
```

goreleaser is useful. :smile:

**Comment by kristofferahl** [comment]
Ah, I see.. I didn't look enough at the code obviously. 😄 I can see why this is good to have! Merging!

**Pull request merged**